capsules/src/nrf51822_serialization.rs

```diff
@@ -270,14 +315,22 @@ impl uart::ReceiveClient for Nrf51822Serialization<'_> {
                 // hardware, regardless of how much space (if any) was
                 // available in the buffer provided by the app.
                 kernel_data.schedule_upcall(0, (4, rx_len, len)).ok();
-            });
+            }) {
```

Comment:
Something is weird with this line. As I'm reading it, the `})` ends the closure for `self.apps.enter()`. But then it's followed by an opening `{`?? I expected there to be an `} else {` like the above code has.

Am I just misunderstanding something?

Reply:
The open `{` is the bracket for the `if` and its enclosed block is if the grant entry had an error.

Reply:
Oh geez. Okay, that parses now.